geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -381,9 +381,7 @@ public boolean virtualPut(EntryEventImpl event, boolean ifNew, boolean ifOld,
         // if the listeners have already seen this event, then it has already
         // been successfully applied to the cache. Distributed messages and
         // return
-        if (isTraceEnabled) {
-          logger.trace("DR.virtualPut: this cache has already seen this event {}", event);
-        }
+        logger.debug("DR.virtualPut: this cache has already seen this event {}", event);
```

Comment:
Was moving this out of an if check intentional?

Reply:
yes, I want to see that log entry at debug level.   Debug level for the regression test I'm running already spits out 35gb of logs.  I hate to think what trace level would do.

Reply:
You should keep any logger.debug statements gated by "if (logger.isDebugEnabled())". Geode wraps Log4J2 loggers with the FastLogger class in Geode to short-circuit the more expensive filter/log-level checking that occurs in Log4J2. If you don't gate the log statements then the code path will be slowed-down quite measurably by Log4j2. Trace and debug level statements should be gated.